Dockerfile

```diff
@@ -15,16 +15,73 @@
 # along with Invenio; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 
+###############################################################################
+## 1. Base (stable)                                                          ##
+###############################################################################
+
 FROM python:2.7
+MAINTAINER CERN <info@invenio-software.org>
+
+# nodejs repo
+# detects distribution and adds the right repo
+# See: https://github.com/joyent/node/wiki/Installing-Node.js-via-package-manager#debian-and-ubuntu-based-linux-distributions
+RUN curl -sL https://deb.nodesource.com/setup | bash -
 
+# install requirements from repos
 RUN apt-get update && \
-    apt-get -qy install ssl-cert poppler-utils git subversion --fix-missing
+    apt-get -qy install poppler-utils git subversion nodejs mysql-client sudo --fix-missing
+
+# install python requirements
+RUN pip install --upgrade pip && \
+    pip install mock && \
+    pip install unittest2 && \
+    pip install watchdog
+
+# install nodejs requirements
+RUN npm update && \
+    npm install --silent -g bower less clean-css uglify-js requirejs
+
+# add invenio user
+RUN useradd --home-dir /home/invenio --create-home --shell /bin/bash invenio
+
 
-# Adding current directory as `/code`.
+###############################################################################
+## 2. Code (changing)                                                        ##
+###############################################################################
+
+# select proper requierments level
+#ENV REQUIREMENTS lowest
+ENV REQUIREMENTS release
+#ENV REQUIREMENTS devel
+
+# add current directory as `/code`.
 ADD . /code
 WORKDIR /code
```

Comment:
Adding whole code base at this point seems too soon.  It would invalidate the docker build cache even if only docs files change in Invenio.  It is better to copy only pip requirements files and friends at this point, and install those, as we do in other packages.  This has a consequence that, unless requirements change, the docker can reuse prior layers, and so the new docker image will be built in no time.  (without running pip repetitively)